lib/vintage_net_lte/provider.ex

```diff
@@ -0,0 +1,34 @@
+defmodule VintageNetLTE.Provider do
```

Comment:
I'm struggling with the name `Provider`. I know it's more typing, but `ServiceProvider` seems more clear. There might be other words. What do you think?

Reply:
My other thought on naming is that this is probably going to end up being a Service Provider/Plan lookup since a service provider can have you use a different APN based on your plan.

Reply:
I like `ServiceProvider`, should we also change the configuration for the provider to be `:service_provider`? 

I have this module super simple for now and figured we can extend later to provide more information to the lookup.